### ISSUE TITLE
Set `run-on` as User Preference

### ISSUE THREAD

**Issue by rm3l** [redhat-developer/odo#6331]
Exposing as part of the --run-on flag is useful if someone wants to switch between the two modes. Have we considered also putting it as part of the preference so that the user doesn't have to specify that on every odo dev or odo deploy command?

_Originally posted by @elsony in https://github.com/redhat-developer/odo/issues/6151#issuecomment-1285454465_

**Comment by rm3l** [comment]
Response: https://github.com/redhat-developer/odo/issues/6151#issuecomment-1302167437

> > Exposing as part of the --run-on flag is useful if someone wants to switch between the two modes. Have we considered also putting it as part of the preference so that the user doesn't have to specify that on every odo dev or odo deploy command?
> 
> That makes sense. We can add this. But it is something that we should consider only after we have a more solid implementation and it is ready to be moved out of experimental mode.

**Comment by valaparthvi** [comment]
I have set a low priority since this should be implemented after odo is more stable on podman.
> But it is something that we should consider only after we have a more solid implementation and it is ready to be moved out of experimental mode. https://github.com/redhat-developer/odo/issues/6151#issuecomment-1302167437
